tests/unit/test_finder.py

```diff
@@ -105,12 +105,22 @@ def test_finder_detects_latest_already_satisfied_pypi_links():
     req.satisfied_by = satisfied_by
     finder = PackageFinder(
         [],
-        ["http://pypi.org/simple/"],
+        ["https://pypi.org/simple/"],
         session=PipSession(),
     )
+    finder_get_pages = finder._get_pages
 
-    with pytest.raises(BestVersionAlreadyInstalled):
-        finder.find_requirement(req, True)
+    def get_pages(locations, project_name):
+        assert locations == [Link('https://pypi.org/simple/initools/')]
+        for page in finder_get_pages(locations, project_name):
+            get_pages.page_found = True
+            yield page
+    get_pages.page_found = False
```

Comment:
Would it be more pragmatic to use the mock’s `side_effect` for this, instead of mocking the function with a custom one? That way you can also add an `assert finder._get_pages.call_count == 1` to make the check more obvious.